schemas/glean/glean/glean.1.schema.json

```diff
@@ -136,7 +136,7 @@
                   "type": "integer"
                 },
                 "propertyNames": {
-                  "pattern": "[0-9]+"
+                  "pattern": "^[0-9]+$"
```

Comment:
Do we know which regex engine is used for this check? Because `^` and `$` usually mean `line-start` and `line-end`, thus if the data contains a newline it could still be invalid (plus the fact that a newline-containing value is wrong anyway in our case)

Reply:
JSON Schema specifies the [ECMA 262 regex syntax](https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.6.3.3).  In that case, `^` and `$` are beginning and end of input, respectively, unless the [multiline flag is set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Assertions), but I can only assume that it isn't for JSON Schema (and experimentation confirms that)...